### ISSUE TITLE
1Torch was not compiled with flash attention.

### ISSUE THREAD

**Issue by ialhabbal** [VectorSpaceLab/OmniGen#88]
I installed the latest version of pytorch and confirmed installation 2.4.1+cu124, when I ran an image generation I got the following message:

:\OmniGen\venv\lib\site-packages\transformers\models\phi3\modeling_phi3.py:633: UserWarning: 1Torch was not compiled with flash attention. (Triggered internally at C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\aten\src\ATen\native\transformers\cuda\sdp_utils.cpp:555.)
  attn_output = torch.nn.functional.scaled_dot_product_attention(

How do I enable flash attention?

**Comment by staoxiao** [comment]
@ialhabbal, FlashAttention does not support arbitrary forms of attention masks, so OmniGen cannot use FlashAttention at the moment.

**Comment by ItsNoOneReally** [comment]
I also have this problem.
I installed OmniGen via pinocio.computer and this happens every time I try to generate anything.
@staoxiao 
I don't really care if it can/can not use flash attention i just want it to work (ie make a generation).
Could you explain how to stop this error?

**Comment by mxoutkast** [comment]
experienced the same problem whenever I generate something this warning shows up and generation is stuck at 0% progress. I'm using torch==2.3.1+cu118

**Comment by Ares1983** [comment]
I have the same warning and the generation is extremely slow (about 3 minutes per single step).
